File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/ChampionDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.Champion;

import java.util.List;

@Dao
public interface ChampionDao {

    @Query("SELECT * FROM champions ORDER BY name")
    LiveData<List<Champion>> getChampions();

    @Query("SELECT * FROM champions ORDER BY name")
    List<Champion> getChampionsWidget();

    @Query("SELECT * FROM champions WHERE `key` = :championKey")
    LiveData<Champion> getChampionByKey(int championKey);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllChampion(List<Champion> champions);

    @Query("DELETE FROM champions")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateChampion(Champion champion);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/ChampionDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.Champion

@Dao
interface ChampionDao {
    @Query("SELECT * FROM champions ORDER BY name")
    fun champions(): LiveData<List<Champion>>

    @Query("SELECT * FROM champions ORDER BY name")
    fun championsWidget(): List<Champion>

    @Query("SELECT * FROM champions WHERE `key` = :championKey")
    fun getChampionByKey(championKey: Int): LiveData<Champion>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllChampion(champions: List<Champion>)

    @Query("DELETE FROM champions")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateChampion(champion: Champion)
}